### PR TITLE
Add frontend for fuzzing

### PR DESCRIPTION
**Pull request by josephbisch**
Use the following configure command:

$ ./configure --with-fuzzer --with-fuzzer-lib=/path/to/libFuzzer.a \
      CC=clang CXX=clang++

Places an irssi-fuzz in src/fe-fuzz/ after build.

Also can specify SANFLAGS to override the chosen sanitizer flags
(defaults to "-g -fsanitize=address -fsanitize-coverage=trace-pc-guard").

### CODE REVIEW

File: src/fe-fuzz/module-formats.c [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/*
```
Comment (ailin-nemui) [inline src/fe-fuzz/module-formats.c]: I wonder if we could #include the other formats instead of having top copy*paste?

File: configure.ac [hunk 2]
```diff
@@ -611,6 +650,7 @@ echo
 
 echo "Building text frontend ........... : $want_textui"
 echo "Building irssi bot ............... : $want_irssibot"
+echo "Building irssi fuzzer ............ : $want_irssifuzzer"
```
Comment (ahf) [inline configure.ac]: I think this is a slightly internal thing, so I don't think we should "confuse" the average user by making them wonder what "irssi fuzzer" is if they give compiling irssi for themselves a try.

File: src/fe-fuzz/Makefile.am [hunk 3]
```diff
@@ -0,0 +1,25 @@
+bin_PROGRAMS = irssi-fuzz
+
+# Force link with clang++ for libfuzzer support
+CCLD=clang++ $(CXXFLAGS)
```
Comment (ahf) [inline src/fe-fuzz/Makefile.am]: Probably do a check if `clang++` is available if fuzzing is enabled in `configure.ac`?

File: src/fe-fuzz/irssi.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#include "modules-load.h"
+#include "levels.h"
+#include "../fe-text/module-formats.h" // need to explicitly grab from fe-text
+#include "themes.h"
+#include "core.h"
+#include "fe-common-core.h"
+#include "args.h"
+#include "printtext.h"
+
+#include <stdbool.h>
```
Comment (ahf) [inline src/fe-fuzz/irssi.c]: Is this needed? It's a C99 feature and I don't think we build irssi as a C99 application right now - we might, but I'm not sure.

**Comment by josephbisch** [comment]
Okay, fixed nits other than clang++ check.

Let me know if I should squash commits.

**Comment by ahf** [comment]
LGTM.

**Pull request merged**